util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,181 @@
+    #
+    # @example
+    #   initialize(results_dir, prefix)
+    #
+    def initialize(results_dir, prefix, json2graphite_path)
+      @results_dir = results_dir
+      @prefix = prefix
+      @json2graphite_path = json2graphite_path
+      @id = @results_dir.split("_").last
+    end
+
+    # The main entry point to the import_metrics_files script
+    #
+    # @author Bill Claytor
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files
+      pmc_dir = "#{@results_dir}/puppet-metrics-collector"
+      raise "Directory not found: #{pmc_dir}" unless File.directory? pmc_dir
+
+      puts "Importing puppet_metrics_collector files:"
+      puts " results_dir: #{@results_dir}"
+      puts " prefix: #{@prefix}"
+      puts " id: #{@id}"
+
+      service_dirs = Dir.glob("#{pmc_dir}/*").select { |f| File.directory? f }
+
+      puts " service directories:"
+      puts service_dirs
+      puts
+
+      # each service directory
+      service_dirs.each do |service_dir|
+        puts "Checking directory: #{service_dir}"
+
+        # each host directory
+        host_dirs = Dir.glob("#{service_dir}/*").select { |f| File.directory? f }
+
+        puts " host directories: "
+        puts host_dirs
+        puts
+
+        host_dirs.each do |host_dir|
+          import_metrics_files_for_host_dir(host_dir)
+        end
+      end
+    end
+
+    # Calls the json2graphite.rb script for the specified host directory
+    # with a server tag using the following pattern: prefix_id_hostname
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] host_dir The host directory to import
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files_for_host_dir(host_dir)
+      hostname = File.basename host_dir
+      server_tag = "#{@prefix}_#{@id}_#{hostname}"
+      pattern = "'#{host_dir}/*.json'"
+      cmd = "ruby #{@json2graphite_path} --pattern #{pattern}" \
+        " --convert-to influxdb --netcat localhost --influx-db puppet_metrics --server-tag #{server_tag}"
+      puts "Importing puppet-metrics-collector files for host: #{hostname}"
+      puts " cmd: #{cmd}"
+      puts
+
+      `#{cmd}`
+    end
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
+
+  DEFAULT_JSON2GRAPHITE_PATH = File.expand_path "~/git/puppet-metrics-viewer/json2graphite.rb"
+
+  DESCRIPTION = <<~DESCRIPTION
+    This script imports metrics collector data for the Standard and Large reference architectures.
+    It checks each service subdirectory of the puppet-metrics-collector directory and calls the json2graphite.rb script once for each host subdirectory found within.
+    The server tags are constructed using the following pattern:
+    prefix_id_hostname
+
+    For example:
+    be ruby util/metrics/import_metrics.rb -r results/scale/PERF_SCALE_12345 -p slv-649
+
+    This will result in the following server tags:
+    slv-649_12345_127.0.0.1
```

Comment:
doesn't get 127.0.0.1 anymore right... so should make this match a normal hostname

Reply:
Updated